src/common/layers/Layer.js

```diff
@@ -117,13 +119,29 @@ export default class Layer extends Observable {
     this.terminate();
     /** @ignore */
     this.map = map;
+    if (!this.map || !this.olLayer) {
+      return;
+    }
+
+    this.olListenersKeys.push(
+      this.map.getLayers().on('remove', (evt) => {
```

Comment:
you are in the common layer , there should not be reference to mapbox or ol function, move it to src/ol/layers

Reply:
Obviously you are right :+1: I correct it